scripts/dts/extract/globals.py

```diff
@@ -331,6 +331,51 @@ def build_cell_array(prop_array):
 
     return ret_array
 
+def nexus_to_controller(cell_parent, gpio_index):
+    # Aim of this function is to return a gpio-controller
+    # from the cell_parent provided which has been identified as a 'nexus node'
+    # (ie has a gpio-map property).
+    # thanks to the gpio_index provided.
+    # Note: Nexus nodes and gpio-map property are described in devide tree
+    # specification, v0.3, chapter 'Nexus nodes and Specifier Mapping'.
+
+    # Assumption 1: For now, we assume otionnal properties gpio-map-mask and
+    # gpio-map-pass-thru are not specified
+
+    # First, retrieve gpio-map as a list
+    raw_map = reduced[cell_parent]['props']['gpio-map']
+
+    # We now need to build the map as an array of array,
+    # which will allow to use provided index
+
+    # map raws are encoded as follows:
+    # [child specifier][gpio controller phandle][parent specifier]
+    # child specifier field length is connector property #gpio-cells
+    # parent specifier field length is parent property #gpio-cells
+
+    child_specifier_size = reduced[cell_parent]['props']['#gpio-cells']
+    # Assumption 2: for now we assume parent #gpio-cells is constant across
+    # the map, so we take the first one and apply to the whole map
+    parent = phandles[child_specifier_size+1]
+    parent_specifier_size = reduced[parent]['props']['#gpio-cells']
+    array_cell_size = child_specifier_size + 1 + parent_specifier_size
+
+    # Now that raws length is known, build the map as an array of arrays
+    index = 0
+    map_array = []
+    while index < len(raw_map):
+        map_array.append(raw_map[index:index+array_cell_size])
+        index += array_cell_size
+
+    # Return the gpio controller matching the provided index
+    try:
+        # controller phandle is to be found in the map, using gpio_index
+        controller_phandle = map_array[gpio_index][child_specifier_size]
+        return phandles[controller_phandle]
```

Comment:
Can't immediately tell which of the `[gpio_index]`, `[child_specifier_size]`, and `[controller_phandle]` lookups are meant to be able to generate the `IndexError` under "normal" operation of the program here.

Better to do something like this, and avoid `try`/`except` completely:

```python
if gpio_index not in map_array:
    return None

...
```

That makes the intention clear, and means that errors in the code won't be eaten by the `except IndexError:`.